### PR TITLE
[hw/ip] Updated ADC_CTRL testplan

### PR DESCRIPTION
**Pull request by NigelScales**

### CODE REVIEW

**Comment by NigelScales** [comment]
> Since this overall looks good to designer, we should schedule to go over it.
> 
> Another general comment: we recently adopt this format in the testplan, like this [one](https://github.com/lowRISC/opentitan/blob/master/hw/ip/keymgr/data/keymgr_testplan.hjson). This may make the testplan more structural, but this is not a hard requirement.
> 
> ```
> one-line description
> Stimulus:
> - S1
> - S2
> 
> Checks:
> - C1
> - C2
> ```
I think I can split it like that if you like. 

It's not really feasible to include all of the checks in each test entry as they are generally part of the environment and apply to all tests.
That's why I've abbreviated them to e.g.  "predict when filters should be triggered".
In practice this will be done with a model of the filters - probably as part of the scoreboard.
It looks like the place for describing this in detail will be in here:
https://github.com/opentitan/hw/ip/adc_ctrl/doc/dv/index.md
I expect the description of the checks will go in sections for the scoreboard and assertions.

**Comment by NigelScales** [comment]
> @NigelScales will you guys be extending this to top level testing as well? Because there are very specific cases I have in mind.
> 
> Basically the way this will get used in a chip is that we will first detect a set of filter conditions to trigger either a wakeup / interrupt. If the device is asleep, it should trigger a wakeup, resume clocks and then allow the interrupt to propagate. It might be good to simulate this case where the normal clock "stops".
> 
> Once we're in this state, we then we want to detect the inverse condition (ie, outside the range) as either a wakeup / interrupt. Again the same statements above apply.
> 
> We then we want to detect the inside condition again, basically we're going to toggle between in/out/in/out, and the wakeup/interrupt case must always both work.
> 
> Does that make sense?

I'll add a test in V3 similar to the rest test but with main clock off during the detection phase.
This will ensure the filters etc. are running from the always on clock with no CDC interlocks.

**Comment by weicaiyang** [comment]
> > Since this overall looks good to designer, we should schedule to go over it.
> > Another general comment: we recently adopt this format in the testplan, like this [one](https://github.com/lowRISC/opentitan/blob/master/hw/ip/keymgr/data/keymgr_testplan.hjson). This may make the testplan more structural, but this is not a hard requirement.
> > ```
> > one-line description
> > Stimulus:
> > - S1
> > - S2
> > 
> > Checks:
> > - C1
> > - C2
> > ```
> 
> I think I can split it like that if you like.
> 
> It's not really feasible to include all of the checks in each test entry as they are generally part of the environment and apply to all tests. That's why I've abbreviated them to e.g. "predict when filters should be triggered". In practice this will be done with a model of the filters - probably as part of the scoreboard. It looks like the place for describing this in detail will be in here: https://github.com/opentitan/hw/ip/adc_ctrl/doc/dv/index.md I expect the description of the checks will go in sections for the scoreboard and assertions.

make sense. The current format looks good to me.

**Review by sriyerg — APPROVED** [review]
Skimmed through, LGTM overall. Will rely on @tjaychen's feedback on content.

File: hw/ip/adc_ctrl/data/adc_ctrl_testplan.hjson [hunk 1]
```diff
@@ -12,22 +12,279 @@
...
+
+          - From monitored ADC data predict when filters should be triggered.
+          - Confirm this by reading the filter status register.
+          - Check wakeup signal operates as expected.
+          '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: filters_wakeup_fixed
```
Comment (weicaiyang) [inline hw/ip/adc_ctrl/data/adc_ctrl_testplan.hjson:138]: @sriyerg 

These several filter tests are very similar. After the first one, should we only capture the differences in the other entries to reduce the duplication? WDYT?

Comment (sriyerg) [reply hw/ip/adc_ctrl/data/adc_ctrl_testplan.hjson:138]: Agreed, but fine either way. Testpoints to refer to other testpoints for description if similar stimulus / checks apply.

Comment (weicaiyang) [reply hw/ip/adc_ctrl/data/adc_ctrl_testplan.hjson:138]: Thanks Sri

**Comment by weicaiyang** [comment]
@NigelScales we have done the testplan review earlier this week. Please update the testplan as discussed. Then we can merge it.

**Review by weicaiyang — COMMENTED** [review]
@NigelScales 
From the review notes, looks like we haven't addressed these 2 items.

> [Sri] Add a one-hot assertion to check the two channels requests in design.
> [Weicai] Please add stress_all and stress_all_with_rand_reset to the adc_ctrl testplan and remove the filters_both test.

**Review by msfschaffner — APPROVED** [review]
LGTM but defer to @weicaiyang to merge it.

**Review by weicaiyang — APPROVED** [review]

**Pull request merged**